### ISSUE TITLE
[FR] Add OpenCL support

### ISSUE THREAD

**Issue by mark-stopka** [ergoplatform/cuda-miner#67]
Currently the miner supports just NVIDIA CUDA, I think as the official miner, it should support both CUDA and OpenCL used by Intel and AMD...

There is an [AMD miner](https://github.com/mhssamadani/ergoAMDminer), as a non-official project, but I believe the two should be merged...

**Comment by mark-stopka** [comment]
@kushti what do you think?

**Comment by mhssamadani** [comment]
you're right. such thing will be done for the new miner due to incoming fork.
